scrapy/http/response/json.py

```diff
@@ -0,0 +1,16 @@
+"""
+This module implements the JsonResponse class which adds encoding
+discovering through JSON encoding declarations to the TextResponse class.
+
+See documentation in docs/topics/request-response.rst
+"""
+
+import json
+from scrapy.http.response.text import TextResponse
+
+
+class JsonResponse(TextResponse):
+
+    def json(self):
```

Comment:
Also, I wonder if it would be appropriate to do some encoding handling. For instance, see `python-requests`'s implementation: https://github.com/psf/requests/blob/v2.23.0/requests/models.py#L874-L898

Reply:
Good point. As far as I understand the code encoding is already handled in the base class `TextResponse`.

Reply:
Right, but AFAICT that handles the body conversion from bytes to unicode, I wonder if there are some considerations to be made when decoding the unicode body to JSON. Perhaps there is actually nothing to be done in this regard, in any case we can always fix any issues later.